vowpalwabbit/shared_feature_merger.cc

```diff
@@ -0,0 +1,112 @@
+#include "shared_feature_merger.h"
+#include "cb.h"
+#include "example.h"
+#include "label_dictionary.h"
+#include "learner.h"
+#include "options.h"
+#include "parse_args.h"
+#include "vw.h"
+
+#include <iterator>
+
+namespace VW
+{
+namespace shared_feature_merger
+{
+static const std::vector<std::string> option_strings = {
```

Comment:
Have an explicitly enumerated list is kind of subpar because it will inevitably fail to be updated properly in the future. 

Can we instead test directly for whether or not the parent reduction is multiline?

Reply:
Unless I missed something, there's no way to know which other reductions are being used in the setup function unless we pass it through the all object. Since that would require a similar level of maintenance, I opted for options checking where we could at least encapsulate the check in a single place

Reply:
I think we should leave this for a later refactoring.